### ISSUE TITLE
Referencing the action starts an elastic container build step even if the action itself is skipped

### ISSUE THREAD

**Issue by porgabi** [elastic/elastic-github-actions#35]
The Elasticsearch action is conditionally called in a workflow, but even when it's not run, the container build step still runs. Here's an [example](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/10082467596/job/27876788040#step:2:1) for such a build:
![image](https://github.com/user-attachments/assets/2cbab0ff-f785-44d0-b7d5-c156bae9c3d2)

The Elasticsearch action is referenced in the action.yml file found in [this pull request](https://github.com/Lombiq/GitHub-Actions/pull/376/files), which is conditionally called in the build-and-test-orchard-core.yml file.

Is this normal behavior, or can it be prevented somehow? The container build step shouldn't run when the Elasticsearch setup step doesn't.

**Comment by Piedone** [comment]
Anybody with tips on how to prevent this Docker build step that happens after the job setup if we don't actually use the action?

**Comment by pquentin** [comment]
Did it actually run? The / symbol suggests that this step was skipped.

**Comment by Piedone** [comment]
No, that's the point. The build step runs even if the action doesn't.
